src/ducks/apps/components/ApplicationRouting.jsx

```diff
@@ -1,28 +1,34 @@
 import React, { Component } from 'react'
 import { Route, withRouter } from 'react-router-dom'
 
+import { InstallModal } from '../currentAppVersion/Containers'
+
 import ApplicationDetails from './ApplicationDetails'
 import UninstallModal from './UninstallModal'
 
 export class ApplicationRouting extends Component {
   render () {
-    const { installedApps, isFetching, parent, history } = this.props
+    const { apps, installedApps, isFetching, isInstalling, parent, history } = this.props
+    const appsArray = apps || installedApps || []
     return (
       <div>
         <Route path={`/${parent}/:appSlug`} render={({ match }) => {
           if (isFetching) return
-          if (installedApps.length && match.params) {
-            const app = installedApps.find(app => app.slug === match.params.appSlug)
+          if (appsArray.length && match.params) {
+            const app = appsArray.find(app => app.slug === match.params.appSlug)
             if (!app) return history.push(`/${parent}`)
-            return <ApplicationDetails app={app} parent={parent} />
+            return <ApplicationDetails app={app} parent={parent} installed={app.installed} />
```

Comment:
Why do you need to pass `app.installed` since we already have this information in `app` already passed as props.